applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h

```diff
@@ -0,0 +1,1015 @@
+
+//         #pragma omp parallel
+//         {
+            // Create a tree
+            // It will use a copy of mNodeList (a std::vector which contains pointers)
+            // Copying the list is required because the tree will reorder it for efficiency
+            KDTree TreePoints(mPointListOrigin.begin(), mPointListOrigin.end(), mBucketSize); 
+            
+            // Iterate over the destination elements
+            ElementsArrayType& pElements = mrDestinationMainModelPart.Elements();
+            auto numElements = pElements.end() - pElements.begin();
+            
+//             #pragma omp for
+            for(unsigned int i = 0; i < numElements; i++) 
+            {
+                auto itElem = pElements.begin() + i;
+                
+                // Getting the geometry
+                Element::GeometryType& rThisGeometry = itElem->GetGeometry();
+                
+                // Getting the integration points
+                ThisIntegrationMethod = itElem->GetIntegrationMethod();
+                const Element::GeometryType::IntegrationPointsArrayType& IntegrationPoints = rThisGeometry.IntegrationPoints(ThisIntegrationMethod);
+                const unsigned int IntegrationPointsNumber = IntegrationPoints.size();
+                
+                // Getting the CL
+                std::vector<ConstitutiveLaw::Pointer> ConstitutiveLawVector(IntegrationPointsNumber);
+                itElem->GetValueOnIntegrationPoints(CONSTITUTIVE_LAW,ConstitutiveLawVector,CurrentProcessInfo);
+            
+                for (unsigned int iGaussPoint = 0; iGaussPoint < IntegrationPointsNumber; iGaussPoint++ )
+                {
+                    // We compute the global coordinates
+                    const array_1d<double, 3> LocalCoordinates = IntegrationPoints[iGaussPoint].Coordinates();
+                    array_1d<double, 3> GlobalCoordinates;
+                    GlobalCoordinates = rThisGeometry.GlobalCoordinates( GlobalCoordinates, LocalCoordinates );
+                    
+                    PointTypePointer pGPOrigin = TreePoints.SearchNearestPoint(GlobalCoordinates);
+                    
+                    for (unsigned int iVar = 0; iVar < mInternalVariableList.size(); iVar++)
+                    {
+                        Variable<double> ThisVar = mInternalVariableList[iVar];
+                        
+                        double OriginValue;
+                        OriginValue = (pGPOrigin->GetConstitutiveLaw())->GetValue(ThisVar, OriginValue);
+                        
+                        (ConstitutiveLawVector[iGaussPoint])->SetValue(ThisVar, OriginValue, CurrentProcessInfo);
+                    }
+                }
+            }
+//         }
+    }
+    
+    /**
+     * This method interpolate the values of the GP using the LST method
+     */
+        
+    void InterpolateGaussPointsLST()
+    {
+        // We Initialize the process info
```

Comment:
it would be good to check here on the first node if you have the variable NODAL_H